### PR TITLE
Unify unittests for WSIReader

### PR DESCRIPTION
**Pull request by drbeh**
### Description
This PR unifies unittests for WSIReader with different backends and also adds new tests to cover situations where WSIReader is being used in a transform and loaded via a data loader.

Related to: https://github.com/Project-MONAI/MONAI/pull/3257#issuecomment-960479226

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests`.

### CODE REVIEW

**Comment by drbeh** [comment]
@Nic-Ma @wyli , I'd appreciate your review here.

File: tests/test_wsireader.py [hunk 1]
```diff
@@ -68,60 +73,100 @@
...
+    backend = None
 
     @parameterized.expand([TEST_CASE_0])
-    def test_read_whole_image(self, file_path, expected_shape):
-        reader = WSIReader("cuCIM")
+    def test_read_whole_image(self, file_path, level, expected_shape):
+        reader = WSIReader(self.backend, level=level)
         img_obj = reader.read(file_path)
         img = reader.get_data(img_obj)[0]
         self.assertTupleEqual(img.shape, expected_shape)
```
Comment (wyli) [inline tests/test_wsireader.py:107]: looks like this class doesn't have the methods such as `self.assertTupleEqual`.

Comment (drbeh) [reply tests/test_wsireader.py:107]: This class are not being called directly so that would be fine but I'll change it.

Comment (drbeh) [reply tests/test_wsireader.py]: update that.

Comment (drbeh) [reply tests/test_wsireader.py]: @wyli, updated the tests. Is it good now?

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me.

**Comment by wyli** [comment]
there's an error after merging this PR, any idea? 

https://github.com/Project-MONAI/MONAI/actions/runs/1428856690

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Thanks for raising the issue.
I am trying to fix it locally.

Thanks.

**Comment by wyli** [comment]
thanks @drbeh , is it related to this one? 
https://github.com/Project-MONAI/MONAI/runs/4125617483?check_suite_focus=true

```
======================================================================
ERROR: test_with_dataloader_0____w_MONAI_MONAI_tests_testing_data_temp_1sGTKZlJBIz53pfqTxoTqiIQzIoEzHLAe_tiff (tests.test_wsireader.TestCuCIM)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/parameterized/parameterized.py", line 533, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/__w/MONAI/MONAI/tests/test_wsireader.py", line 154, in test_with_dataloader
    spatial_shape = tuple(d.item() for d in data["image_meta_dict"]["spatial_shape"])
  File "/__w/MONAI/MONAI/tests/test_wsireader.py", line 154, in <genexpr>
    spatial_shape = tuple(d.item() for d in data["image_meta_dict"]["spatial_shape"])
ValueError: only one element tensors can be converted to Python scalars

----------------------------------------------------------------------
```

**Comment by Nic-Ma** [comment]
Hi @wenqi ,

This error message is related to my PR https://github.com/Project-MONAI/MONAI/pull/3255, already fixed in the latest commit: https://github.com/Project-MONAI/MONAI/pull/3255/commits/d84a39bb61411bdfd129b90ca5af364cbfc3ad3d

Thanks.

**Pull request merged**